cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -322,6 +326,20 @@ type partitionState struct {
 	// UnlockKey was what key the partition was unlocked with, either "run",
 	// "fallback" or "recovery".
 	UnlockKey string `json:"unlock-key,omitempty"`
+
+	// unexported internal fields for tracking the device, these are used during
+	// state machine execution, and then combined into Device during finalize()
+	// for simple representation to the consumer of degraded.json
+
+	// decryptedDevice is what decrypted mapper device corresponds to the
+	// partition, it can have the following states
+	// - successfully decrypted => the decrypted mapper device
+	// - unencrypted => the block device of the partition
+	// - identified as decrypted, but failed to decrypt => empty string
+	decryptRawDevice string
```

Comment:
this should be called decryptedDevice

Reply:
hmm now unclear that we should continue to call it decryptedDevice, but I will push up regardless

Reply:
I think we should use partDevice and fsDevice here too (?)

Reply:
works for me